File: app/src/main/kotlin/erikterwiel/dhmodestroyer/BTStateReceiver.java
package erikterwiel.dhmodestroyer;

import android.bluetooth.BluetoothAdapter;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.Log;

/**
 * Created by Erik on 10/23/2017.
 */

public class BTStateReceiver extends BroadcastReceiver {

    private final String TAG = "BTStateReceiver.java";

    // Starts BondActivity when Bluetooth is turned on
    @Override
    public void onReceive(Context context, Intent intent) {

        Log.i(TAG, "onReceive() called");

        SharedPreferences senderDatabase =
                context.getSharedPreferences("senderDatabase", Context.MODE_PRIVATE);

        BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        String action = intent.getAction();
        if (action.equals(bluetoothAdapter.ACTION_STATE_CHANGED)) {
            int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, bluetoothAdapter.ERROR);
            switch (state) {
                case BluetoothAdapter.STATE_TURNING_OFF:
                    Log.i(TAG, "Bluetooth is turning off");
                    break;
                case BluetoothAdapter.STATE_OFF:
                    Log.i(TAG, "Bluetooth is now off");
                    break;
                case BluetoothAdapter.STATE_TURNING_ON:
                    Log.i(TAG, "Bluetooth is turning on");
                    break;
                case BluetoothAdapter.STATE_ON:
                    Log.i(TAG, "Bluetooth is now on");
                    if (senderDatabase.getString("toOpen", null).equals("Bond")) {
                        Log.i(TAG, "Bluetooth enabled, launching BondActivity");
                        Intent bondIntent = new Intent(context, BondActivity.class);
                        context.startActivity(bondIntent);
                    } else if (senderDatabase.getString("toOpen", null).equals("Connect")) {
                        Log.i(TAG, "Bluetooth enabled, launching ConnectActivity");
                        Intent connectIntent = new Intent(context, ConnectActivity.class);
                        context.startActivity(connectIntent);
                    }
                    break;
            }
        }
    }
}

File: app/src/main/kotlin/erikterwiel/dhmodestroyer/BTStateReceiver.kt
package erikterwiel.dhmodestroyer

import android.bluetooth.BluetoothAdapter
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BTStateReceiver : BroadcastReceiver() {

    private val TAG = "BTStateReceiver.kt"

    // Starts BondActivity when Bluetooth is turned on
    override fun onReceive(context: Context, intent: Intent) {

        Log.i(TAG, "onReceive() called")

        val senderDatabase = context.getSharedPreferences("senderDatabase", Context.MODE_PRIVATE)

        val action = intent.action
        if (action == BluetoothAdapter.ACTION_STATE_CHANGED) {
            val state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR)
            when (state) {
                BluetoothAdapter.STATE_TURNING_OFF -> Log.i(TAG, "Bluetooth is turning off")
                BluetoothAdapter.STATE_OFF -> Log.i(TAG, "Bluetooth is now off")
                BluetoothAdapter.STATE_TURNING_ON -> Log.i(TAG, "Bluetooth is turning on")
                BluetoothAdapter.STATE_ON -> {
                    Log.i(TAG, "Bluetooth is now on")
                    if (senderDatabase.getString("toOpen", null) == "Bond") {
                        Log.i(TAG, "Bluetooth enabled, launching BondActivity")
                        val bondIntent = Intent(context, BondActivity::class.java)
                        context.startActivity(bondIntent)
                    } else if (senderDatabase.getString("toOpen", null) == "Connect") {
                        Log.i(TAG, "Bluetooth enabled, launching ConnectActivity")
                        val connectIntent = Intent(context, ConnectActivity::class.java)
                        context.startActivity(connectIntent)
                    }
                }
            }
        }
    }
}
